src/Raven.Server/Utils/CertificateUtils.cs

```diff
@@ -186,8 +186,21 @@ public static X509Certificate2 CreateSelfSignedExpiredClientCertificate(string c
                     new X509KeyUsage(X509KeyUsage.KeyCertSign | X509KeyUsage.CrlSign));
             }
 
-            // Serial Number
-            var serialNumber = new BigInteger(20 * BitsPerByte, random);
+            // Serial Number - we must generate a positive number, so retrying if it's not
+            BigInteger serialNumber;
+            var retries = 10;
+
+            while (true)
+            {
+                var serialNumberBytes = new byte[20];
+                random.NextBytes(serialNumberBytes);
+                serialNumber = new BigInteger(serialNumberBytes);
+                if (serialNumber.SignValue > 0)
```

Comment:
Huh? Why do we need this like that?

Reply:
If the generated bytes form a negative number, certificateGenerator.SetSerialNumber() will throw

Reply:
Then why regenerate the number? flip the sign

https://github.com/bcgit/bc-csharp/blob/master/crypto/src/math/BigInteger.cs#L721

Reply:
Ooh that's nice, I missed this method.